src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx

```diff
@@ -162,9 +162,9 @@ export const EVMHomeAdaptorProvider = ({
     });
 
     wallet?.provider?.on("accountsChanged", (accounts: string[])=> {
-      console.log("Accounts changed:", { account, accounts });
-      if (walletSelected) setWalletType("unset");
-      setAccount(accounts[0])
+      console.log("Accounts changed:", { walletSelected, account, accounts });
+      if (walletSelected && account && account !== accounts[0]) setWalletType("unset");
```

Comment:
Could you assign the whole expression to a variable with a good name?

Reply:
Fixed